ctapipe/reco/FitGammaHillas.py

```diff
@@ -0,0 +1,308 @@
+    tel_dir = linalg.set_phi_theta(tel_phi,tel_theta)
+
+    pix_dirs = []
+    
+    
+    for a, b in zip(pix_alpha,pix_beta):
+        pix_dir = linalg.set_phi_theta( tel_phi, tel_theta + b )
+        
+        pix_dir = linalg.rotate_around_axis(pix_dir, tel_dir, (a-camera_rotation))
+        pix_dirs.append(pix_dir*u.dimless)
+        
+    return pix_dirs
+
+
+class FitGammaHillas(RecoShowerGeomAlgorithm):
+    
+    def __init__(self):
+        self.tel_geom = {}
+        self.circles = {}
+    
+    def setup_geometry(self, telescopes, cameras, optics, phi=180.*u.deg, theta=20.*u.deg):
+        self.Ver = 'Feb2016'
+        self.TelVer = 'TelescopeTable_Version{}'.format(self.Ver)
+        self.CamVer = 'CameraTable_Version{}_TelID'.format(self.Ver)
+        self.OptVer = 'OpticsTable_Version{}_TelID'.format(self.Ver)
+        
+        self.telescopes = telescopes[self.TelVer]
+        self.cameras    = lambda tel_id : cameras[self.CamVer+str(tel_id)]
+        self.optics     = lambda tel_id : optics [self.OptVer+str(tel_id)]
+    
+        self.tel_phi   = phi
+        self.tel_theta = theta
+
+    def predict(self, Hillas_dict, seed_pos=[0,0] ):
+        self.get_great_circles(Hillas_dict)
+        dir1 = self.fit_origin_crosses()[0]
+        dir2 = self.fit_origin_minimise(dir1)
+        
+        pos = self.fit_core(seed_pos)
+        
+        result = RecoShowerGeom("FitGammaHillas")
+        (phi, theta) = linalg.get_phi_theta(dir2)
+        #TODO make sure az and phi turn in same direction...
+        result.alt, result.az = theta-90*u.deg, phi
+        result.core_x = pos[0]
+        result.core_y = pos[1]
+        
+        result.tel_ids = [h for h in Hillas_dict.keys()]
+        
+        result.is_valid = True
+        
+        result.average_size    = np.mean([ h.size for h in Hillas_dict.values() ])
+        
+        result.alt_uncert      = -1.
+        result.az_uncert       = -1.
+        result.core_uncert     = -1.
+        result.h_max           = -1.
+        result.h_max_uncert    = -1.
+        result.goodness_of_fit = -1.
+        
+        
+        
+        return result
+        
+    def get_great_circles(self, Hillas_dict):
+        self.circles = {}
+        for tel_id, moments in Hillas_dict.items():
+
+            
+            camera_rotation = -90.*u.deg
+            #if tel_id in TelDict["LST"]          : camera_rotation = -110.893*u.deg
+
+            
+            circle = GreatCircle(guessPixDirection( np.array([ moments.cen_x, (moments.cen_x + moments.length * np.cos( moments.psi + np.pi/2 ))] ) * u.m,
+                                                    np.array([ moments.cen_y, (moments.cen_y + moments.length * np.sin( moments.psi + np.pi/2 ))] ) * u.m,
+                                                    self.tel_phi, self.tel_theta, self.telescopes['FL'][tel_id-1] * u.m, camera_rotation=camera_rotation
+                                                  )
+                                )
```

Comment:
still some code that goes well over 80 columns. Can you either reformat this or break it into 2 steps (e.g. define the 2 `np.arrays` and then pass them to `guess_pix_direction`?  The 80 column rule makes it much easier later to review and merge changes by looking at two pieces of code side-by-side, and forces peoepl to not use too complex lines of code.  In Python, you also don't need to use a continuation line separator if code is between parentheses, for example:

```python
guess_pix_direction(self.tel_phi,
                    self.tel_theta,
                    self.telescopes..)
```
is valid code